pkg/registries/topo.go

```diff
@@ -49,6 +49,12 @@ func (g *topoGraph) getNode(node topoNode) *internalTopoNode {
 	return res
 }
 
+// AddNode creates and adds a node to the graph if it does not already exist.
+// If the node already exists, this function does nothing.
+func (g *topoGraph) AddNode(node topoNode) {
```

Comment:
Unit tests of `topoGraph` adding an isolated node to an existing connected graph, and testing a single-node graph, would be nice.

At the very least, the “we use `topoGraph` only for ordering known-connected components” comment should be updated.